lib/plugins.js

```diff
@@ -55,9 +52,36 @@ Plugins.prototype.loadPlugins = function () {
       plugin = isBuiltIn;
     }
 
-    gateway.addPlugin(pluginName, _.bind(plugin.init, plugin));
+    config.proxies.forEach(function (proxy) {
+        gateway.addPlugin(proxy, pluginName, _.bind(plugin.init, plugin));
+    })
   });
 
+    config.proxies.forEach(function (proxy) {
+      if (proxy.plugins) {
+        proxy.plugins.forEach(function (p) {
+            var pluginName = Object.keys(p)[0];
+
+            const isBuiltIn = builtInPlugins[pluginName];
+            let plugin;
+            if (!isBuiltIn) {
+                const segment = path.join(pluginDir, pluginName);
+                // look for the plugins dir one level up from the lib dir (this file's location) if not absolute
+                const pluginFullPath = path.isAbsolute(segment)
+                    ? segment
+                    : path.resolve(__dirname, path.join('..', segment));
+                debug('loading plugin from ' + pluginFullPath);
+                plugin = require(pluginFullPath);
+            } else {
+                plugin = isBuiltIn;
+            }
+
+
+          gateway.addPlugin(proxy, pluginName, _.bind(plugin.init, plugin));
```

Comment:
I understand that this change brings a per-proxy selection of which plugins to run, but not a per-proxy configuration (or incremental configuration) of each plugins. That would be desirable.